docs/authoring/endpoints/pagination.md

```diff
@@ -0,0 +1,41 @@
+# Pagination
+The support of pagination is data sources is given by following template:
+```
+{
+     "endpointUrl": “someUrl”
+     "callbackContextTemplate": "" //set of parameters that need to be used in the next call
+     "callbackRequiredTemplate": "boolean" //template decides if further calls need to be made to service
+     "initialContextTemplate": "" //defines the initial set of query params
+}
+```
+
+Important fields explained:
+- `endpointUrl` : Standard context here. The url that you intend to call.
+- `callbackContextTemplate` : This is the set of parameters that we will need to call this API again. Generally, this will be derived from the response of the previous call. It's supposed to be a set of key-value pairs.
+- `callbackRequiredTemplate` : This is basically a boolean flag which identifies whether further call to API is needed. It does so based on the response of the previous call.
+- `initialContextTemplate` : This is the set of parameters needed for the first call. It is also a set of key-value pairs.
+
+Depending on the kind of pagination supported by the underlying API, the logic to determine the value of "callbackRequiredTemplate" can be anything. 
+We expect both "callbackContextTemplate" and "callbackRequiredTemplate" to be provided as mustache templates. Examples follow:
+
+### Sample Data Source
```

Comment:
Add examples of how 2 well know formats can be supported?

Reply:
I will just add the link of the doc which explains this. Explaining that here would be redundant I think.